### PR TITLE
Add auth helpers, and v15 support

### PR DESCRIPTION
**Pull request by mcardosos**
Thanks for your contribution! Please let us know what problems this PR addresses or what functionality it adds. Please link to other related issues and PRs to speed up review. Thank you!

### CODE REVIEW

**Comment by vladbarosan** [comment]
> 	err := helpers.LoadEnvVars()

isnt this supposed to load the env variables ? why do we laod the afterwards

---
Refers to: iam/oauth.go:60 in 6f5cc77. [](commit_id = 6f5cc771d50dddd9f95fc3b86fa3c13af3013bf2, deletion_comment = False)

**Comment by vladbarosan** [comment]
> }

instead of this cant we use [https://github.com/Azure/go-autorest/blob/cd2702d0255126f3d64a7cef703e1341f22bc0ac/autorest/azure/auth/auth.go#L247](https://github.com/Azure/go-autorest/blob/cd2702d0255126f3d64a7cef703e1341f22bc0ac/autorest/azure/auth/auth.go#L247) ?

---
Refers to: iam/oauth.go:185 in 6f5cc77. [](commit_id = 6f5cc771d50dddd9f95fc3b86fa3c13af3013bf2, deletion_comment = False)

**Comment by mcardosos** [comment]
> 	err := helpers.LoadEnvVars()

After where?

---
In reply to: [378392473](https://github.com/Azure-Samples/azure-sdk-for-go-samples/pull/161#issuecomment-378392473) [](ancestors = 378392473)

---
Refers to: iam/oauth.go:60 in 6f5cc77. [](commit_id = 6f5cc771d50dddd9f95fc3b86fa3c13af3013bf2, deletion_comment = False)

**Comment by mcardosos** [comment]
> 	err := helpers.LoadEnvVars()

Oh, LoadEnvVars just reads the `.env` file. Will rename, it is somewhat ambiguous.

---
In reply to: [378395895](https://github.com/Azure-Samples/azure-sdk-for-go-samples/pull/161#issuecomment-378395895) [](ancestors = 378395895,378392473)

---
Refers to: iam/oauth.go:60 in 6f5cc77. [](commit_id = 6f5cc771d50dddd9f95fc3b86fa3c13af3013bf2, deletion_comment = False)

File: iam/oauth.go [hunk 1]
```diff
@@ -195,37 +177,23 @@ func GetKeyvaultToken(grantType OAuthGrantType) (authorizer autorest.Authorizer,
 
 	switch grantType {
 	case OAuthGrantTypeServicePrincipal:
-		spt, err := adal.NewServicePrincipalToken(
-			*config,
-			clientID,
-			clientSecret,
-			"https://vault.azure.net")
-
+		token, err := adal.NewServicePrincipalToken(*config, clientID, clientSecret, vaultEndpoint)
```
Comment (vladbarosan) [inline iam/oauth.go]: why isn't this one using the same auth mechanism ?

Comment (mcardosos) [reply iam/oauth.go]: It needs a different token endpoint

**Comment by vladbarosan** [comment]
Ok looks good. :shipit:

**Pull request merged**